.github/workflows/analyse.yml

```diff
@@ -0,0 +1,28 @@
+name: analyse
+
+on: ['push', 'pull_request']
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: PHPStan
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.2
+          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, exif, iconv, imagick, fileinfo
+          coverage: none
+
+      - name: Repository Credentials
+        run: composer config http-basic.repo.magento.com ${{ secrets.MAGENTO_USERNAME }} ${{ secrets.MAGENTO_PASSWORD }}
```

Comment:
While this works great, i think using
https://fooman.com/blog/no-authentication-needed-magento-2-mirror.html

Would make it easier to run phpstan in ones own repository, without having to go after credentials